src/lib/dapp/index.ts

```diff
@@ -180,16 +163,18 @@ export class DAppConnector {
    * @param pairingTopic - The pairing topic for the connection (optional).
    * @returns A Promise that resolves when the connection process is complete.
    */
-  public async openModal(pairingTopic?: string): Promise<void> {
-    const { uri, approval } = await this.connectURI(pairingTopic)
-    const walletConnectModal = new WalletConnectModal({
-      projectId: this.projectId,
-      chains: this.supportedChains,
-    })
-    walletConnectModal.openModal({ uri })
-    const session = await approval()
-    await this.onSessionConnected(session)
-    walletConnectModal.closeModal()
+  public async openModal(pairingTopic?: string): Promise<SessionTypes.Struct> {
+    try {
+      const { uri, approval } = await this.connectURI(pairingTopic)
+      this.walletConnectModal.openModal({ uri })
+      const session = await approval()
+      await this.onSessionConnected(session)
+      return session
+    } catch (error) {
+      throw error
```

Comment:
You can use a `try...finally`, omitting the catch block since nothing is being done in this catch block.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#the_finally_block